addons/milight.rb

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/ruby
+require 'socket'
+require 'json'
+
+module Milight
+  def self.config()
+      milight_config = File.expand_path(
+                       File.join(
+                         ENV['JANE_PATH'], 'config', 'milight.json'
+                       )
+                     )
+      return JSON.parse(File.read(milight_config), symbolize_names: true)
+    end
+
+  # master list of commands here: http://www.limitlessled.com/dev/
+
+  WIFI_BRIDGE_IP = config[:wifi_bridge_ip]
+  WIFI_BRIDGE_PORT = config[:wifi_bridge_port]
+
+  ######## GROUP 1 ########
+  #define lamp states
+  LAMP_ON = "\x45\x00\x55"
+  LAMP_OFF = "\x46\x00\x55"
+
+  #define white
+  WHITE_INIT = "\x45\x00\x55"
+  WHITE_FINISH = "\xC5\x00\x55"
+
+  #define max brightness
+  MAX_BRIGHT = "\x4E\x1B\x55"
+
+  #define max dim
+  MAX_DIM = "\x4E\x02\x55"
+
+  #define disco mode
+  DISCO = "\x4D\x00\x55"
+
+  #define speed states
+  SPEED_UP = "\x44\x00\x55"
+  SPEED_DOWN = "\x43\x00\x55"
+
+  #define night mode
+  NIGHT_MODE = "\xC6\x00\x55"
+
+  def self.on()
+    socket = UDPSocket.new
```

Comment:
you can make your life a lot easier by either caching this or putting it into a helper function:

```
def self.socket
   UDPSocket.new
end

def self.other 
    socket.send ...
end

```

Not sure my ruby is correct here, but you are repeating yourself over and over ;)